### PR TITLE
GIX-1686: SnsNeuronAdvancedSection actions

### PR DESCRIPTION
**Pull request by lmuntaner**
# Motivation

We are shuffling the data in the neuron detail to make it more user friendly.

In this PR: Add the action items in the sns neuron advanced section.

# Changes

* Render the auto stake maturity and split button in the SnsNeuronAdvancedSection.

# Tests

* Add test to check that actions are present.

# Todos

Not yet worth an entry in the changelog.

### CODE REVIEW

**Comment by lmuntaner** [comment]
New UI:

Screenshot 2023-07-24 at 08 32 06

**Comment by lmuntaner** [comment]
@dskloetd @mstrasinskis please review

**Comment by dskloetd** [comment]
> Not yet worth an entry in the changelog.

Isn't that what the "Not Published" section is for? You don't need an entry for each of these PRs, but one entry about ongoing work in the neuron settings makes sense, I think. Otherwise, let's just remove the "Not Published" section, or am I misunderstanding what it is for?

**Comment by dskloetd** [comment]
The checkbox in the screenshot is basically invisible. Shouldn't it have a different background color?

File: frontend/src/lib/components/sns-neuron-detail/SnsNeuronAdvancedSection.svelte [hunk 1]
```diff
@@ -83,5 +91,8 @@
     flex-direction: column;
     align-items: flex-start;
     gap: var(--padding-2x);
+
+    --checkbox-padding: 0;
+    --checkbox-label-order: 1;
```
Comment (dskloetd) [inline frontend/src/lib/components/sns-neuron-detail/SnsNeuronAdvancedSection.svelte:96]: I really hope the new Figma process will get rid of stuff like this.

File: frontend/src/lib/pages/SnsNeuronDetail.svelte [hunk 2]
```diff
@@ -156,7 +156,7 @@
           <SkeletonCard cardType="info" separator />
           <SkeletonCard cardType="info" separator />
         {:else}
-          {#if $ENABLE_NEURON_SETTINGS && nonNullish(parameters) && nonNullish(token) && nonNullish($selectedSnsNeuronStore.neuron)}
+          {#if $ENABLE_NEURON_SETTINGS && nonNullish(parameters) && nonNullish(token) && nonNullish($selectedSnsNeuronStore.neuron) && nonNullish(transactionFee)}
```
Comment (dskloetd) [inline frontend/src/lib/pages/SnsNeuronDetail.svelte:159]: I guess all these checks are just to satisfy the type system?
And it doesn't work if it gets replaced with a single variable that combines all these checks? Like
```
$: areRequiredParamsPresent = nonNullish(parameters) && nonNullish(token) && nonNullish($selectedSnsNeuronStore.neuron) && nonNullish(transactionFee);
```

**Review by dskloetd — APPROVED** [review]

File: frontend/src/lib/components/sns-neuron-detail/SnsNeuronAdvancedSection.svelte (hunk 1 above)
Comment (lmuntaner) [reply frontend/src/lib/components/sns-neuron-detail/SnsNeuronAdvancedSection.svelte:96]: This is not just Figma, but how our components work. I also don't like it to be honest.

File: frontend/src/lib/pages/SnsNeuronDetail.svelte (hunk 2 above)
Comment (lmuntaner) [reply frontend/src/lib/pages/SnsNeuronDetail.svelte:159]: Yes, I can do that. I kept adding one at a time, but I believe it makes sense now to put them all in one variable.

Comment (lmuntaner) [reply frontend/src/lib/pages/SnsNeuronDetail.svelte:159]: Damn, TS doesn't like this... It doesn't understand that if the variables is true, then each type is present.

Comment (lmuntaner) [reply frontend/src/lib/pages/SnsNeuronDetail.svelte:159]: Maybe we can think of a better way

File: frontend/src/lib/components/sns-neuron-detail/SnsNeuronAdvancedSection.svelte (hunk 1 above)
Comment (dskloetd) [reply frontend/src/lib/components/sns-neuron-detail/SnsNeuronAdvancedSection.svelte:96]: But if we have a consistent design, we don't need to do custom configuration each time we use a checkbox. The default checkbox would just be according to the default design.

**Comment by dskloetd** [comment]
> I wanted to wait until it was all finished. Which is almost there.

If we cut a release now, there won't be a change log entry but there will be unpublished work.
So I think it makes more sense to add an entry with the first of several PRs than with the last of several PRs.

File: frontend/src/lib/components/sns-neuron-detail/SnsNeuronAdvancedSection.svelte (hunk 1 above)
Comment (lmuntaner) [reply frontend/src/lib/components/sns-neuron-detail/SnsNeuronAdvancedSection.svelte:96]: That's also true. But I also think that it shouldn't have padding by default. And I can also see that the design might cover checkboxes left and right. Although ideally not...

**Pull request merged**